diffpy/snmf/io.py

```diff
@@ -0,0 +1,107 @@
+      signal weightings change smoothly with time. Assumes a default value of 1e18.
+
+    Returns
+    -------
+    dictionary
+      The collection of the names and values of the constants used in the algorithm. Contains the number of observed PDF
+      /XRD patterns, the length of each pattern, the type of the data, the number of components the user would like to
+      decompose the data into, an initial guess for the component matrix, and initial guess for the weight factor matrix
+      ,an initial guess for the stretching factor matrix, a parameter controlling smoothness of the solution, a
+      parameter controlling sparseness of the solution,
+
+    """
+    signal_length = data_input.shape[0]
+    moment_amount = data_input.shape[1]
+
+    component_matrix_guess = np.random.rand(signal_length, component_amount)
+    weight_matrix_guess = np.random.rand(component_amount, moment_amount)
+    stretching_matrix_guess = np.ones(component_amount, moment_amount) + np.random.randn(component_amount,
+                                                                                         moment_amount) * 1e-3
+
+    diagonals = [np.ones(moment_amount - 2), -2 * np.ones(moment_amount - 2), np.ones(moment_amount - 2)]
+    sparsity_term = .25 * scipy.sparse.diags(diagonals, [0, 1, 2], shape=(moment_amount - 2, moment_amount))
+
+    return {
+        "signal_length": signal_length,
+        "moment_amount": moment_amount,
+        "component_matrix_guess": component_matrix_guess,
+        "weight_matrix_guess": weight_matrix_guess,
+        "stretching_matrix_guess": stretching_matrix_guess,
+        "component_amount": component_amount,
+        "data_type": data_type,
+        "smoothness": smoothness,
+        "sparsity": sparsity,
+        "sparsity_term": sparsity_term
+
+    }
+
+
+def load_input_signals(file_path=None):
+    """Processes a directory of a series of PDF/XRD patterns into a usable format.
+
+    Constructs a 2d array out of a directory of PDF/XRD patterns containing each files dependent variable column in a
+    new column. Constructs a 1d array containing the independent variable values.
+
+    Parameters
+    ----------
+    file_path: str or Path object, optional
+      The path to the directory containing the input data. If no path is specified, defaults to the current working
+      directory.  Accepts a string or a pathlib.Path object.
+
+    Returns
+    -------
+    tuple
+      The output containing a 2d array containing a PDF/XRD pattern as each of its columns and a 1d array containing the
+      independent variable values of the PDF/XRD pattern series.
+
+    """
+
+    if file_path is None:
+        directory_path = Path.cwd()
+    else:
+        directory_path = Path(file_path)
+
+    values_list_to_array = []
+    grid_list_to_array = []
+    for pattern_path in directory_path.glob('*'):
+
+        if pattern_path.is_file():
```

Comment:
I think the inputs here will be .gr and .iq files and similar.  Please use `loaddata` from `diffpy.utils` (I htink) rather than trying to recode it all.